### ISSUE TITLE
Make it work in Blue Ocean

### ISSUE THREAD

**Issue by deiga** [jenkinsci/ansicolor-plugin#108]
For Blue Ocean this plugin only removes the color codings but does not output color coded strings

### Test environment

- Jenkins Version: 2.46.2
- AnsiColor Version: 0.5.0
- Hostsystem: Debian
- Blue Ocean Version: 1.0.1

### Expected behavior

Colored lines in Blue Ocean Console log

### Actual behavior

Lines are missing color code strings but still show as default color

### Steps to reproduce the behavior

Open build inside Blue Ocean and look at build log

**Comment by mkobit** [comment]
https://issues.jenkins-ci.org/browse/JENKINS-41200 is open for Blue Ocean support

**Comment by tnguyen14** [comment]
@mkobit looks like that JENKINS-41200 issue is resolved?

**Comment by ashb** [comment]
I think BlueOcean supports colouring, but not specifically the ansi-color plugin which would need https://issues.jenkins-ci.org/browse/JENKINS-46324 (if the distinction matters to anyone.)

When I last looked at the jenkins issues it looked as if:

- If you have ansi-color in your pipeline you get colour in your classic log, but not in Blue
- If you don't have ansi-color you end up with color in Blue, but _not_ in the classic log.

(This is all just based on the Jira tickets, I haven't actually tested this so I might be waay off)

:(

**Comment by kpritam** [comment]
@ashb  yes this is exactly how it works.

**Comment by ssbarnea** [comment]
Same bug is hosted at https://issues.jenkins-ci.org/browse/JENKINS-39536
I wonder if there is a way to fix it.

**Comment by rinu** [comment]
When a pipeline is used is it possible to get color in the email for a build failure? Without this plugin I can see color in Blue but it's garbage like "[37;41mMyText [0m" in the email. Seems impossible until this issue is resolved.

**Comment by reinholdfuereder** [comment]
https://github.com/jenkinsci/ansicolor-plugin/pull/132 appears to (by and large?) solve this

**Comment by jglick** [comment]
It should, yes—BO does its own (relatively simple) handling of ANSI escape sequences, which as of #132 are not being stripped from the raw log.

**Comment by killerwizz1** [comment]
Works beautifully in BlueOcean with 0.6.2.
Only issue I seem to have is that "Show complete log" I see the escape sequence. We have a handful of "long" stages that overflows that would be nice. This is the /blue/rest output. 
URL would be something along this line:
https://jenkins-server/blue/rest/organizations/jenkins/pipelines/Folder/pipelines/Jobs/branches/master/runs/1/nodes/44/steps/206/log/?start=0

**Comment by dblock** [comment]
Let's close this!

**Comment by aramse** [comment]
I've yet to find any combination of configurations that makes this work in BlueOcean. I'm using BlueOcean 1.16.0 and ansicolor 0.6.2. The BlueOcean output _always_ strips the ansi color, while the traditional Jenkins console output _never_ does. Also, console output does actually render the color properly with `options {ansiColor('xterm')}' in the Jenkinsfile. Anyone else seeing similar behavior?

**Comment by KenDJohnson** [comment]
> I've yet to find any combination of configurations that makes this work in BlueOcean. I'm using BlueOcean 1.16.0 and ansicolor 0.6.2. The BlueOcean output _always_ strips the ansi color, while the traditional Jenkins console output _never_ does. Console output renders the color properly with `options {ansiColor('xterm')}` in the Jenkinsfile but BlueOcean still strips so no luck there. Anyone else seeing similar behavior?

@aramse or anyone else looking for an answer to this:

I was running into the same issue and could not figure out why it wasn't showing the colors in BlueOcean. I have builds that do, but this specific one wouldn't show. After comparing some the differences in the raw pipeline logs, I found that the successful ones use `\e[1;31merror\e[0m` for a bold red error, while the one without colors was `\e[1;38;5;9merror\e[0m` which is IIRC bold red using the 8-bit/256 colors. I switched the formatting in the troublesome build and it worked properly. It looks like while the logs/console output elsewhere in Jenkins handle 256 bit color, BlueOcean does not.

tl;dr
If your colors aren't showing up, see if you can use the simple colors in the 30-37 range rather than the 256 bit `;` delimited sequences.
